testsuite/tests/generic-optional-arguments/test_non_value_constructors.ml

```diff
@@ -0,0 +1,37 @@
+(* TEST
+ flags = "-extension-universe alpha";
+ expect;
```

Comment:
This needs native tests since unboxed types are represented as boxed in bytecode.